### ISSUE TITLE
Pillifying removes character before Pill

### ISSUE THREAD

**Issue by t3chguy** [vector-im/riot-web#4855]
so `(Travis` `[TAB]` results in Pill but `(` goes away :(

**Comment by lukebarnard1** [comment]
This seems unique to `\S` characters, that is characters that are not considered whitespace. And this only happens when tab completing, so

```js
if (force && this.shouldForceComplete()) {
        commandRegex = /\S+/g;
}
```

is looking suspect.

This means that when tab-completing, the regex changes from `@\S*` to `\S+` for the UserProvider (the thing providing user completions). `\S+` will go and match anything before the display name entered that isn't whitespace. 

This is a side-effect of tab-completing free-form text to free-form, arbitrary display names.

**Comment by ara4n** [comment]
fixed in matrix-org/matrix-react-sdk#1890
